pkg/eraser/eraser.go

```diff
@@ -0,0 +1,248 @@
+)
+
+var (
+	// Timeout  of connecting to server (default: 10s)
+	Timeout time.Duration
+)
+
+func GetAddressAndDialer(endpoint string) (string, func(ctx context.Context, addr string) (net.Conn, error), error) {
+	protocol, addr, err := parseEndpointWithFallbackProtocol(endpoint, unixProtocol)
+	if err != nil {
+		return "", nil, err
+	}
+	if protocol != unixProtocol {
+		return "", nil, fmt.Errorf("only support unix socket endpoint")
+	}
+
+	return addr, dial, nil
+}
+
+func dial(ctx context.Context, addr string) (net.Conn, error) {
+	return (&net.Dialer{}).DialContext(ctx, unixProtocol, addr)
+}
+
+func parseEndpointWithFallbackProtocol(endpoint string, fallbackProtocol string) (protocol string, addr string, err error) {
+	if protocol, addr, err = parseEndpoint(endpoint); err != nil && protocol == "" {
+		fallbackEndpoint := fallbackProtocol + "://" + endpoint
+		protocol, addr, err = parseEndpoint(fallbackEndpoint)
+	}
+	return
+}
+
+func parseEndpoint(endpoint string) (string, string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", "", err
+	}
+
+	switch u.Scheme {
+	case "tcp":
+		return "tcp", u.Host, nil
+
+	case "unix":
+		return "unix", u.Path, nil
+
+	case "":
+		return "", "", fmt.Errorf("using %q as endpoint is deprecated, please consider using full url format", endpoint)
+
+	default:
+		return u.Scheme, "", fmt.Errorf("protocol %q not supported", u.Scheme)
+	}
+}
+
+func getImageClient(context *cli.Context) (pb.ImageServiceClient, *grpc.ClientConn, error) {
+	addr, dialer, err := GetAddressAndDialer("unix:///run/containerd/containerd.sock")
+
+	if err != nil {
+		fmt.Print("get address and dialer err")
+	}
+
+	conn, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithTimeout(Timeout), grpc.WithContextDialer(dialer))
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	imageClient := pb.NewImageServiceClient(conn)
+
+	return imageClient, conn, nil
+}
+
+func ListImages(client pb.ImageServiceClient, image string) (resp *pb.ListImagesResponse, err error) {
+	request := &pb.ListImagesRequest{Filter: &pb.ImageFilter{Image: &pb.ImageSpec{Image: image}}}
+
+	resp, err = client.ListImages(context.Background(), request)
+
+	return
+}
+
+func RemoveImage(client pb.ImageServiceClient, image string) (resp *pb.RemoveImageResponse, err error) {
+	if image == "" {
+		return nil, fmt.Errorf("ImageID cannot be empty")
+	}
+
+	request := &pb.RemoveImageRequest{Image: &pb.ImageSpec{Image: image}}
+
+	resp, err = client.RemoveImage(context.Background(), request)
+
+	return
+}
+
+func contains(slice []string, str string) bool {
+	for _, s := range slice {
+		if s == str {
+			return true
+		}
+	}
+	return false
+}
+
+func removeDuplicateValues(intSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+
+	for _, entry := range intSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+
+func main() {
```

Comment:
Let's move all the app logic out of `main()`.
I tend to keep `main()` for setting up things related to setting up the process: flags, log config, and general bootstrapping. Then we can move app logic to something that returns an `error` which we can handle in `main()` (such as print to stderr and exit non-zero)